storage/diemdb/src/schema/transaction_by_hash/mod.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module defines physical storage schema for a transaction index via which the version of
+//! of a transaction sent by `transaction_hash` can be found. With the version one can resort to
+//! `TransactionSchema` for the transaction content.
+//!
+//! ```text
+//! |<--key-->|<-value->|
+//! |   hash  | txn_ver |
+//! ```
+
+use crate::schema::{ensure_slice_len_eq, TRANSACTION_BY_ACCOUNT_CF_NAME};
+use anyhow::Result;
+use byteorder::{BigEndian, ReadBytesExt};
+use diem_crypto::HashValue;
+use diem_types::transaction::Version;
+use schemadb::{
+    define_schema,
+    schema::{KeyCodec, ValueCodec},
+};
+use std::mem::size_of;
+
+define_schema!(
+    TransactionByHashSchema,
+    Key,
+    Version,
+    TRANSACTION_BY_ACCOUNT_CF_NAME
+);
+
+type Key = HashValue;
```

Comment:
Seems not necessary? (Or make type alias "Value" as well? This feels inconsistent.)